### PR TITLE
Add `ingest_batch_size` to limit memory usage

### PR DESCRIPTION
**Pull request by idreyn**
We are having a problem ingesting GTFS data on AWS Lambda because this library has a bad habit of loading huge CSVs directly into memory. This PR introduces an `ingest_batch_size` argument to `GtfsFeed::build_locally()` that limits how many CSV rows will be read at once. With the new default of `ingest_batch_size=300000` we cap out at 2GB instead of 10GB, which will ensure this can run reliably in a lambda.

## Testing and profiling

I used the [`memray`](https://bloomberg.github.io/memray/flamegraph.html) memory profiling tool. I added a test script (`scratch.py`) in the root of the repo for the profiler to execute against:

```py
# scratch.py
from mbta_gtfs_sqlite import MbtaGtfsArchive

archive = MbtaGtfsArchive("./feeds")
feed = archive.get_latest_feed()
feed.build_locally()
```

Then I ran the script using memray:

```
poetry shell
python -m memray run -o profile.bin -f -m scratch
```

I inspected the generated feed databases (in my case, in `./feeds/20230819`) and spot-checked them for correctness. Then I generated a memray [flamegraph](https://bloomberg.github.io/memray/flamegraph.html) from the profiler output:

```
python -m memray flamegraph profile.bin  -o profile.html -f
```

_Before:_
Screen Shot 2023-08-27 at 11 09 47 AM

_After:_
Screen Shot 2023-08-27 at 11 09 37 AM

### CODE REVIEW

**Review by austinhouck — APPROVED** [review]
LGTM

File: mbta_gtfs_sqlite/ingest.py [hunk 1]
```diff
@@ -98,36 +99,43 @@ def ingest_rows(
     feed_info: FeedInfo,
     rows: Iterable[Dict[str, str]],
     transforms: RowTransforms = {},
-    individually: bool = False,
+    batch_size: Union[None, int] = None,
```
Comment (mathcolo) [inline mbta_gtfs_sqlite/ingest.py]: May as well just require a batch size?

File: mbta_gtfs_sqlite/ingest.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from sqlalchemy.orm import Session
-from typing import Dict, Any, Callable, List, Type, Iterable
+from typing import Dict, Any, Callable, List, Type, Iterable, Union
+from more_itertools import ichunked
```
Comment (mathcolo) [inline mbta_gtfs_sqlite/ingest.py:3]: TIL more_itertools. very nice

**Review by mathcolo — APPROVED** [review]
Just one question

File: mbta_gtfs_sqlite/ingest.py (hunk 1 above)
Comment (idreyn) [reply mbta_gtfs_sqlite/ingest.py]: Yeah, good call.

**Pull request merged**